### PR TITLE
fix: keep authentication secrets out of logs

### PR DESCRIPTION
**Pull request by QinyingShi**
## Summary

- prevent routine request and response logging for authentication, identity, session, and profile endpoints
- replace raw WeChat codes, OpenIDs, Google OAuth state, email addresses, phone numbers, and names in manual logs with low-cardinality security events
- keep sensitive request size enforcement compatible with Flask versions before and after the per-request limit setter

## Security impact

Passwords, verification codes, session tokens, OAuth state, contact details, device/session metadata, and learner profile data are no longer copied into routine application logs. Sensitive responses also receive `Cache-Control: no-store`.

## Validation

- 113 focused HTTP body-policy, authentication, session, profile, device, Google OAuth, password, and rate-limit tests passed
- Ruff check and format passed
- repository-wide pre-commit checks passed
- no real SMS, email, WeChat, or Google provider calls were made

---

<a href="https://app.devin.ai/review/ai-shifu/ai-shifu/pull/2908" target="_blank"><picture><source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-devin-review-dark.svg?v=4">Devin Review</picture></a>

## Summary by CodeRabbit

- **Bug Fixes**
  - Authentication requests larger than 32 KiB are now rejected consistently across supported Flask versions.
  - Sensitive request and response bodies are omitted from logs, including profile and authentication flows.
  - Logs no longer expose authorization codes, OAuth state, contact details, profile values, or provider-specific sensitive data.
  - Authentication activity now uses clearer, structured event labels.
  - Sensitive endpoints prevent caching of protected request data.
  - Failed external authentication and messaging requests now produce safer, more concise error logs.

- **Tests**
  - Added coverage for request-size limits, cache prevention, body redaction, and sensitive-value exclusion across user flows.

### CODE REVIEW

File: src/api/flaskr/route/user.py [hunk 1]
```diff
@@ -983,6 +1001,7 @@ def get_profile() -> str:
         )
 
     @app.route(path_prefix + "/update_profile", methods=["POST"])
+    @sensitive_body(max_bytes=_AUTH_SENSITIVE_BODY_MAX_BYTES)
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/route/user.py]: Fixed. All canonical learner-profile and profile-onboarding routes now suppress request and response bodies and return Cache-Control: no-store.

File: src/api/flaskr/service/user/utils.py [hunk 2]
```diff
@@ -626,9 +626,9 @@ def _deliver_email(challenge: _PreparedVerificationChallenge) -> bool:
                     server.starttls()
                 server.login(smtp_username, smtp_password)
                 server.sendmail(smtp_sender, email, msg.as_string())
-                app.logger.info("Verification code sent to %s", email)
+                app.logger.info("auth_event=email_verification_code_sent")
             except Exception:
-                app.logger.exception("Failed to send verification code to %s", email)
+                app.logger.exception("auth_event=email_verification_code_failed")
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/service/user/utils.py]: Fixed. SMTP delivery failures now log only the exception type without exception text or a traceback; regression coverage verifies the recipient address is absent.

File: src/api/flaskr/route/user.py [hunk 3]
```diff
@@ -558,6 +561,7 @@ def captcha_api() -> str:
 
     @app.route(path_prefix + "/captcha/verify", methods=["POST"])
     @bypass_token_validation
+    @sensitive_body(max_bytes=_AUTH_SENSITIVE_BODY_MAX_BYTES)
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/route/user.py:560]: Fixed. The CAPTCHA challenge endpoint now suppresses the challenge response from logs and applies Cache-Control: no-store.

File: src/api/flaskr/route/user.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/route/user.py]: Fixed without imposing a new compatibility limit. Sensitive-body suppression is now independent from request-size enforcement, so profile routes remain unbounded by this PR while authentication credential routes retain the 32 KiB limit.

File: src/api/flaskr/route/user.py [hunk 4]
```diff
@@ -83,6 +84,8 @@
 P = ParamSpec("P")
 R = TypeVar("R")
 
+_AUTH_SENSITIVE_BODY_MAX_BYTES = 32 * 1024
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/route/user.py:88]: Fixed. Feedback, avatar, canonical learner-profile, and all profile-onboarding routes now use the sensitive-body policy, with focused route-coverage tests.

File: src/api/flaskr/route/user.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/route/user.py]: Fixed. Profile persistence logs now record only bounded operational metadata such as counts and scope; raw profile values and user-controlled keys are no longer logged.

File: src/api/flaskr/route/user.py [hunk 5]
```diff
@@ -530,14 +536,11 @@ def require_tmp() -> Response:
...
-        if isinstance(wx_code, str) and wx_code:
-            masked_wx_code = f"***{wx_code[-4:]}" if len(wx_code) > 4 else "***"
         app.logger.info(
-            "require_tmp tmp_id: %s, source: %s, wx_code: %s",
-            tmp_id,
+            "auth_event=temp_user_requested has_temp_id=%s source=%s has_wechat_code=%s",
+            bool(tmp_id),
             source,
-            masked_wx_code,
+            bool(wx_code),
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/route/user.py]: Fixed. The temporary-user event now records only whether a source was supplied, never the client-controlled source value.

File: src/api/flaskr/route/user.py [hunk 6]
```diff
@@ -530,14 +536,11 @@ def require_tmp() -> Response:
         source = str(payload.get("source") or "web").strip() or "web"
         wx_code = payload.get("wxcode", None)
         language = payload.get("language") or "en-US"
-        masked_wx_code = None
-        if isinstance(wx_code, str) and wx_code:
-            masked_wx_code = f"***{wx_code[-4:]}" if len(wx_code) > 4 else "***"
         app.logger.info(
-            "require_tmp tmp_id: %s, source: %s, wx_code: %s",
-            tmp_id,
+            "auth_event=temp_user_requested has_temp_id=%s source=%s has_wechat_code=%s",
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/route/user.py]: Fixed. The log event now uses has_source as a boolean and no longer interpolates the untrusted source string.

File: src/api/flaskr/service/user/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/service/user/utils.py]: Fixed. Email delivery failures log only the exception class, and the regression test uses SMTPRecipientsRefused to prove the recipient address is not emitted.

**Comment by QinyingShi** [comment]
Addressed the three follow-up privacy blockers in `a7d944732`:

- WeChat OAuth success logging now emits only an allowlisted event and never serializes access tokens, refresh tokens, OpenID, or UnionID.
- WeChat transport failures are converted at the provider boundary to a stable exception without the request URL, authorization code, or application secret.
- Aliyun SMS rejection and exception logs now retain only the event, provider code, and exception type; phone numbers and provider-controlled response details are omitted.

Validation: 630 user-service tests passed, including new successful-exchange, transport-failure, and recipient-throttling privacy regressions. The focused authentication/SMS suite also passed (73 tests), as did Ruff, formatting, architecture boundaries, UoW checks, and pre-commit hooks. No real provider calls were made.
